ai/src/algorithms/sacd/agent.py

```diff
@@ -78,55 +88,89 @@ def call(self, input_tensor: tf.Tensor) -> Tuple[tf.Tensor, tf.Tensor]:
         return self.actor(input_tensor)
 
     def _copy_target_models(self):
-        for critic_var, target_var in zip(self._critic_1_variables, self._target_critic_1_variables):
-            target_var.assign(self.TARGET_MOMEMTUM * critic_var + (1.0 - self.TARGET_MOMEMTUM) * target_var)
-        for critic_var, target_var in zip(self._critic_2_variables, self._target_critic_2_variables):
-            target_var.assign(self.TARGET_MOMEMTUM * critic_var + (1.0 - self.TARGET_MOMEMTUM) * target_var)
+        for critic_var, target_var in zip(
+            self._critic_1_variables, self._target_critic_1_variables
+        ):
+            target_var.assign(
+                self.TARGET_MOMEMTUM * critic_var
+                + (1.0 - self.TARGET_MOMEMTUM) * target_var
+            )
+        for critic_var, target_var in zip(
+            self._critic_2_variables, self._target_critic_2_variables
+        ):
+            target_var.assign(
+                self.TARGET_MOMEMTUM * critic_var
+                + (1.0 - self.TARGET_MOMEMTUM) * target_var
+            )
 
     def train(self, data):
         state_batch, action_batch, reward_batch, next_state_batch = data
         action_batch = tf.cast(tf.expand_dims(action_batch, 1), tf.float64)
         reward_batch = tf.cast(tf.expand_dims(reward_batch, 1), tf.float64)
 
-        with tf.name_scope('actor_loss'):
+        with tf.name_scope("actor_loss"):
             with tf.GradientTape() as actor_tape:
-                _action, action_probs = self.actor(state_batch)
+                action_probs = self.actor(state_batch)
```

Comment:
This seems like it is a change in behavior? Isn't the first argument from the actor still the action and not the probabilities?

Reply:
Ah, yeah it should be `_, action_probs =`